continuous_integration/environment-3.9.yaml

```diff
@@ -26,6 +26,7 @@ dependencies:
 - pytest-xdist
 - pytest
 - python=3.9
+- py-xgboost=1.7.0
```

Comment:
Noting that we'll need to make a follow-up PR to dask-build-environment as this pinning will definitely conflict with the RAPIDS xgboost.